dom/events/scrolling/wheel-event-groups.html

```diff
@@ -0,0 +1,191 @@
+  tests.forEach((testInfo) => {
+    promise_test(async (t) => {
+      await waitForCompositorCommit();
+
+      // Wait some extra time to ensure nothing from the prior test impacts this one.
+      await new Promise(resolve => step_timeout(resolve, 1000));
+
+      wheelEventTargets = [];
+
+      // Scroll past the boundary of the original element to ensure all events in
+      // group have the same target.
+      await new test_driver.Actions()
+        .addWheel("wheel1")
+        .scroll(testInfo.scrollX, testInfo.scrollY, 0, 30, {origin: testInfo.origin})
+        .pause(1)
+        .scroll(testInfo.scrollX, testInfo.scrollY, 0, 30, {origin: testInfo.origin})
+        .pause(1)
+        .scroll(testInfo.scrollX, testInfo.scrollY, 0, 30, {origin: testInfo.origin})
+        .send();
+
+      await waitForAnimationEnd(() => { return testInfo.scrollingElement.scrollTop; });
+      await waitForCompositorCommit();
+
+      testInfo.scrollingElement.scrollTo({top: 0, left: 0, behavior: "instant"});
+
+      wheelEventTargetsEqual(testInfo.targetElement,
+                             "All wheel events in the group have the same target");
+    }, testInfo.title);
+  });
+
+  promise_test(async (t) => {
+    await waitForCompositorCommit();
+
+    // Wait some extra time to ensure nothing from the prior test impacts this one.
+    await new Promise(resolve => step_timeout(resolve, 1000));
+
+    window.removeEventListener("wheel", pushTargetToWheelEventTargets, {passive: true});
+
+    createSpacerElement('removedSpacer', 'beforebegin', firstRootSpacer);
+
+    wheelEventTargets = [];
+
+    let removedInitialTarget = false;
+
+    // Remove the initial element the wheel event is targetted at once we fire the
+    // first wheel event, then log the subsequent wheel event targets.
+    function onEventRemoveInitialTarget(e) {
+      if (!removedInitialTarget) {
+        removedInitialTarget = true;
+        e.target.remove();
+      } else {
+        wheelEventTargets.push(e.target);
+      }
+    }
+    window.addEventListener("wheel", onEventRemoveInitialTarget, {passive: true});
+
+    await new test_driver.Actions()
+      .addWheel("wheel1")
+      .scroll(40, 2, 0, 30, {origin: "viewport"})
+      .pause(1)
+      .scroll(40, 2, 0, 30, {origin: "viewport"})
+      .pause(1)
+      .scroll(40, 2, 0, 30, {origin: "viewport"})
+      .send();
+
+    await waitForAnimationEnd(() => { return document.scrollingElement.scrollTop; });
+    await waitForCompositorCommit();
+
+    document.scrollingElement.scrollTo({top: 0, left: 0, behavior: "instant"});
+
+    // All subsequent wheel events following the removal of the initial target
+    // should have the same event target.
+    wheelEventTargetsEqual(firstRootSpacer,
+                           "All wheel events in the group have the same target");
+
+    document.removeEventListener("wheel", onEventRemoveInitialTarget, {passive: true});
+  }, "Remove the initial wheel event target.");
```

Comment:
Added a simple test that removes the initial target on the first wheel event. Note that I assumed chromes behavior for this test because it was easier to test. I'm unsure if this is the right behavior.